src/js/utils/dom-data.js

```diff
@@ -3,6 +3,62 @@
  * @module dom-data
  */
 
+import log from './log.js';
+import * as Guid from './guid.js';
+import window from 'global/window';
+
+let FakeWeakMap;
+
+if (!WeakMap) {
+  FakeWeakMap = class {
+    constructor() {
+      this.vdata = 'vdata' + Math.floor(window.performance && window.performance.now() || Date.now());
+      this.data = {};
+    }
+
+    set(key, value) {
+      const access = key[this.vdata] || Guid.newGUID();
```

Comment:
We have a test to make sure the default player ID is still `vjs_video_3`, right? Would hate for that to "regress" again.

Reply:
Ah, yeah, we'd want to double check that on devices without WeakMap. Might require a few more changes.

Reply:
It seems to default to vjs_video_3 in my test just now and the test for default player id is passing on the device.